app/code/Magento/Search/Test/Unit/Ui/Component/Listing/Column/SynonymActionsTest.php

```diff
@@ -0,0 +1,149 @@
+
+        $this->synonymActions = $objectManager->getObject(
+            SynonymActions::class,
+            [
+                'urlBuilder' => $this->urlBuilderMock,
+                'data' => [
+                    'name' => 'actions'
+                ]
+            ]
+        );
+    }
+
+    /**
+     * Test prepareDataSource() with data source has no item
+     */
+    public function testPrepareDataSourceWithNoItem()
+    {
+        $dataSource = [
+            'data' => []
+        ];
+        $expected = [
+            'data' => []
+        ];
+        /**
+         * Assert Result
+         */
+        $this->assertEquals($expected, $this->synonymActions->prepareDataSource($dataSource));
+    }
+
+    /**
+     * Test prepareDataSource() with data source has items
+     */
+    public function testPrepareDataSourceWithItems()
+    {
+        $dataSource = [
+            'data' => [
+                'items' => [
+                    [
+                        'group_id' => self::STUB_SYNONYM_GROUP_ID
+                    ]
+                ]
+            ]
+        ];
+
+        $expected = [
+            'data' => [
+                'items' => [
+                    [
+                        'group_id' => self::STUB_SYNONYM_GROUP_ID,
+                        'actions' => [
+                            'delete' => [
+                                'href' => sprintf(
+                                    self::SYNONYM_GROUP_DELETE_URL,
+                                    self::STUB_SYNONYM_GROUP_ID
+                                ),
+                                'label' => (string)__('Delete'),
+                                'confirm' => [
+                                    'title' => (string)__('Delete'),
+                                    'message' => (string)__(
+                                        'Are you sure you want to delete synonym group with id: %1?',
+                                        self::STUB_SYNONYM_GROUP_ID
+                                    )
+                                ],
+                                '__disableTmpl' => true
+                            ],
+                            'edit' => [
+                                'href' => sprintf(
+                                    self::SYNONYM_GROUP_EDIT_URL,
+                                    self::STUB_SYNONYM_GROUP_ID
+                                ),
+                                'label' => (string)__('View/Edit'),
+                                '__disableTmpl' => true
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+        ];
+
+        $this->urlBuilderMock->method('getUrl')->will(
+            $this->returnValueMap([
+                [
+                    SynonymActions::SYNONYM_URL_PATH_DELETE, ['group_id' => self::STUB_SYNONYM_GROUP_ID],
+                    sprintf(self::SYNONYM_GROUP_DELETE_URL, self::STUB_SYNONYM_GROUP_ID)
+                ],
```

Comment:
According to PSR each element should be in separate line, however I understand the goal of having `2 : 1` set there.